models/weight_recorder.cpp

```diff
@@ -155,16 +158,73 @@ nest::weight_recorder::set_status( const DictionaryDatum& d )
     user_set_precise_times_ = true;
 
   device_.set_status( d );
+
+  P_.set( d );
 }
 
+
 void
 nest::weight_recorder::handle( WeightRecorderEvent& e )
 {
   // accept spikes only if detector was active when spike was
   // emitted
   if ( device_.is_active( e.get_stamp() ) )
   {
+
+    // P_pre_ is defined and sender is not in it
+    if ( !P_.pre_.empty()
+      && std::find( P_.pre_.begin(), P_.pre_.end(), e.get_sender_gid() )
+        == P_.pre_.end() )
+      return;
+
+    // P_post_ is defined and receiver is not in it
+    if ( !P_.post_.empty()
+      && std::find( P_.post_.begin(), P_.post_.end(), e.get_receiver_gid() )
+        == P_.post_.end() )
+      return;
+
+
     WeightRecorderEvent* event = e.clone();
     B_.events_.push_back( *event );
   }
 }
+
+nest::weight_recorder::Parameters_::Parameters_()
+  : pre_()
+  , post_()
+{
+}
+
+nest::weight_recorder::Parameters_::Parameters_( const Parameters_& p )
+  : pre_( p.pre_ )
+  , post_( p.post_ )
+{
+}
+
+void
+nest::weight_recorder::Parameters_::get( DictionaryDatum& d ) const
+{
+  //( *d )[ "pre" ] = pre_;
+  //( *d )[ "post" ] = post_;
+  ( *d )[ "pre" ] = std::vector< long >( pre_.begin(), pre_.end() );
+  ( *d )[ "post" ] = std::vector< long >( post_.begin(), post_.end() );
+}
+
+
+void
+nest::weight_recorder::Parameters_::set( const DictionaryDatum& d )
+{
+
+  if ( d->known( "pre" ) )
+  {
+    std::vector< long > tmp =
+      getValue< std::vector< long > >( d->lookup( "pre" ) );
+    pre_ = std::set< long >( tmp.begin(), tmp.end() );
+  }
+  if ( d->known( "post" ) )
+  {
+    std::vector< long > tmp =
+      getValue< std::vector< long > >( d->lookup( "post" ) );
+    post_ = std::set< long >( tmp.begin(), tmp.end() );
+  }
```

Comment:
Please add `"pre"` and `"post"` to NEST names. And are pre and post really good names here? In all connect routines we use "source" and "target", so that might be better to use here, too, to minimize user confusion.

See  `Parameters_::get()` for comments on vector vs GidCollection.